### PR TITLE
Immutable BatchNorm options

### PR DESCRIPTION
**Pull request by sublee**
Related issue: https://github.com/pytorch/pytorch/issues/20967

This patch makes `BatchNorm`'s `affine` and `track_running_stats` options immutable. They should be strictly determined at the initialization.

### CODE REVIEW

File: torch/nn/modules/batchnorm.py [hunk 1]
```diff
@@ -23,8 +23,8 @@ def __init__(self, num_features, eps=1e-5, momentum=0.1, affine=True,
         self.num_features = num_features
         self.eps = eps
         self.momentum = momentum
-        self.affine = affine
-        self.track_running_stats = track_running_stats
+        self._affine = affine
```
Comment (soumith) [inline torch/nn/modules/batchnorm.py]: i am wondering if this renaming will have implications for loading old state_dicts, didn't look into the code deeply yet

Comment (sublee) [reply torch/nn/modules/batchnorm.py]: Oh, I didn't think about the state_dict migration issue. Thanks for letting me know.

**Comment by sublee** [comment]
@soumith I'm closing this PR to improve backward compatibility. When I reopen it, I'll mention you. Thanks for your code review.
